### PR TITLE
#303 issue resolved : Added students page

### PR DESCRIPTION
**Pull request by Rahul19241**
Added students page with some functionality.
Screenshot 2022-05-27 at 11 21 45 AM
Screenshot 2022-05-27 at 11 21 55 AM

### CODE REVIEW

File: apps/students/models.py [hunk 1]
```diff
@@ -48,6 +55,11 @@ def get_profile_image_url(self):
         else:
             return settings.STATIC_URL + 'home/images/man.png'
 
+    @property
+    def profile_url(self):
+        """Returns the url to volunteer profile."""
```
Comment (divyanshkhetan) [inline apps/students/models.py:60]: please change the comment to 'student profile'

**Review by divyanshkhetan — CHANGES_REQUESTED** [review]
Please make the requested changes in your PR.

File: apps/students/views.py [hunk 2]
```diff
@@ -31,7 +31,11 @@
     login_url=reverse_lazy('accounts:complete_profile')
 )
 def index(request):
-    return HttpResponse('Hello there!')
+    students = Student.objects.order_by('village')
```
[the remark this answers is not in the document]
Comment (Rahul19241) [reply apps/students/views.py]: Done..

File: requirements.txt [hunk 3]
```diff
@@ -33,3 +33,7 @@ social-auth-app-django==4.0.0
 social-auth-core==4.0.3
 sqlparse==0.3.1
 urllib3==1.26.5
+
+
+# SuperUser id : rahulpalahwat@gmail.com
+# password: Rahul@12345
```
[the remark this answers is not in the document]
Comment (Rahul19241) [reply requirements.txt]: Done..

File: static/students/css/style.css [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    /* width: 240px;
+    height: 360px; */
+  }
+  
+  /* .vol__card:focus,
+  .vol__card:hover {
+    transition: all 0.35s cubic-bezier(0.14, 0.51, 0, 1);
+    box-shadow: 10px 10px 0px 0px rgb(239 112 0 / 10%),
+      20px 16px 50px -20px rgb(240 112 0);
+  } */
```
[the remark this answers is not in the document]
Comment (Rahul19241) [reply static/students/css/style.css]: Done

File: templates/students/index.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      <div class="card vol__card mb-5">
+        <!-- <img src="{{ vol.profile.get_profile_image_url }}" class="card-img" alt="Profile Image" /> -->
+        <img src="{{ vol.get_profile_image_url }}" class="card-img" alt="Profile Image" />
+        <div class="card__details">
+          {% if vol.desig %}
+          <span class="tag">{{vol.desig}}</span>
+          {% endif %}
+          <div class="name">{{ vol.get_full_name }}</div>
+          <p>Class : {{vol.school_class}}</p>
+          <p>Village : {{vol.get_village}}</p>
```
[the remark this answers is not in the document]
Comment (Rahul19241) [reply templates/students/index.html]: Done

File: apps/students/models.py (hunk 1 above)
Comment (Rahul19241) [reply apps/students/models.py]: Done

File: templates/students/index.html [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  {% if students %}
+  <div class="row justify-content-center mb-5 pt-5 mx-auto">
+    {% for student in students %}
+    <div class="col-12 col-sm-6 col-md-4 col-lg-3">
+      <div class="card vol__card mb-5">
+        <img src="{{ student.get_profile_image_url }}" class="card-img" alt="Profile Image" />
+        <div class="card__details">
+          {% if student.desig %}
+          <span class="tag">{{student.desig}}</span>
+          {% endif %}
```
Comment (divyanshkhetan) [inline templates/students/index.html:65]: Please remove this code piece.

**Review by divyanshkhetan — CHANGES_REQUESTED** [review]
Please address the changes and resolve the merge conflict in your branch. Thanks!

**Comment by divyanshkhetan** [comment]
@garg3133 LGTM!

**Review by divyanshkhetan — APPROVED** [review]

File: apps/students/models.py [hunk 7]
```diff
@@ -7,6 +7,9 @@
 from home.models import Calendar, Schedule
 
 
+from django.urls import reverse
```
Comment (garg3133) [inline apps/students/models.py:10]: This import statement should be put with the django imports at the top in alphabetical order.

File: apps/students/models.py [hunk 8]
```diff
@@ -36,13 +39,16 @@ class Student(models.Model):
     remarks = models.TextField(max_length=5000, blank=True)
 
     def __str__(self):
-        return f'{self.get_full_name} ({self.school_class})'
+        return f'{self.get_full_name} ({self.school_class}) ({self.get_village}) ({self.get_profile_image_url})'
```
Comment (garg3133) [inline apps/students/models.py:42]: `__str__` method is used to represent the student instance on the admin portal, you don't need to add village and profile_image URL here.

File: apps/students/models.py [hunk 9]
```diff
@@ -36,13 +39,16 @@ class Student(models.Model):
...
 
     @property
     def get_full_name(self):
         return f'{self.first_name} {self.last_name}'
 
     @property
+    def get_village(self):
+        return self.get_village_display()
+
```
Comment (garg3133) [inline apps/students/models.py:51]: We don't really need this property. You can just use `student.get_village_display` in the template file, as you are now using `student.get_village`.

File: apps/students/models.py [hunk 10]
```diff
@@ -36,13 +39,16 @@ class Student(models.Model):
...
     def has_complete_profile(self):
```
Comment (garg3133) [inline apps/students/models.py:52]: You've removed the `@property` tag from here.

File: apps/students/views.py [hunk 11]
```diff
@@ -36,7 +36,11 @@
     login_url=reverse_lazy('accounts:complete_profile')
 )
 def index(request):
-    return HttpResponse('Hello there!')
+    students = Student.objects.order_by('village','school_class','first_name','last_name')
```
Comment (garg3133) [inline apps/students/views.py:39]: We don't need to sort by village, rest are fine. We can later add a filter by village.

File: templates/students/index.html [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+  <div class="fixed-bg">
+    <div id="carouselExampleIndicators" class="carousel slide" data-ride="carousel">
+      <ol class="carousel-indicators">
+        <li data-target="#carouselExampleIndicators" data-slide-to="0" class="active"></li>
+        <li data-target="#carouselExampleIndicators" data-slide-to="1"></li>
+        <li data-target="#carouselExampleIndicators" data-slide-to="2"></li>
+      </ol>
+      <div class="carousel-inner">
+        <div class="carousel-item active">
+          <img class="d-block w-100" src="{% static 'home/images/02.jpg' %}" alt="First slide" style="height:50vh;">
```
Comment (garg3133) [inline templates/students/index.html:24]: I don't think we should keep the make the carousel cover half of the screen. Maybe keep it to just 100-150 px, whatever height looks good. You may also add some media queries for smaller screen sizes.

**Review by garg3133 — COMMENTED** [review]
The carousel looks pretty weird here. Maybe check how the size of the photos is corrected for different screen sizes for the home page header image?

image

File: apps/students/models.py (hunk 7 above)
Comment (Rahul19241) [reply apps/students/models.py:10]: done sir..now it's in alphabetical order

File: apps/students/models.py (hunk 8 above)
Comment (Rahul19241) [reply apps/students/models.py:42]: removes extra fields..

File: apps/students/models.py (hunk 9 above)
Comment (Rahul19241) [reply apps/students/models.py:51]: done sir..

File: apps/students/models.py (hunk 10 above)
Comment (Rahul19241) [reply apps/students/models.py:52]: added sir...

File: apps/students/views.py (hunk 11 above)
Comment (Rahul19241) [reply apps/students/views.py:39]: removes sorting by village..

File: templates/students/index.html (hunk 12 above)
Comment (Rahul19241) [reply templates/students/index.html]: sir for now I have removed the carousel...If in future needed...then we will work on that sir...

**Comment by garg3133** [comment]
Merging, after making a few changes in the last two commits. Thanks @Rahul19241, great work!

Thanks @divyanshkhetan for reviewing this PR.

**Pull request merged**